### PR TITLE
GQL-36: As a MMT dev, I want to be able to CRUD Order Options in CMR GraphQL

### PR DESCRIPTION
**Pull request by trevorlang**
# Overview

### What is the feature?

Adds the ability to create, update, and delete an order option

### What is the Solution?

Adds the relevant mutations

### What areas of the application does this impact?

- `createOrderOption`
- `updateOrderOption`
- `deleteOrderOption`

# Testing

### Reproduction steps

- **Environment for testing:** n/a
- **Collection to test with:** n/a

1. Create an order option per the documentation
2. Update an order option per the documentation
3. Delete an order option per the documentation

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/types/orderOption.graphql [hunk 1]
```diff
@@ -1,57 +1,142 @@
...
-  id: String!
+
   "The orderOption name."
   name: String
-  "The native id to set on the order option."  
+
+  "The native id to set on the order option."
   nativeId: String
-  "The two levels of order options, provider or system"
+
```
Comment (dmistry1) [inline src/types/orderOption.graphql:19]: Can we also add revision-date here. This date is needed in the orderOptions list. 

```
  "A date when the order option was updated"
  revisionDate: String
```

File: src/utils/umm/orderOptionKeyMap.json [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
+    "nativeId"
+  ],
+  "ummKeyMappings": {
+    "associationDetails": "meta.association-details",
+    "conceptId": "meta.concept-id",
+    "description": "umm.Description",
+    "form": "umm.Form",
+    "name": "umm.Name",
+    "nativeId": "meta.native-id",
+    "scope": "umm.Scope",
```
Comment (dmistry1) [inline src/utils/umm/orderOptionKeyMap.json:14]: ` "revisionDate": "meta.revision-date",`

File: src/types/orderOption.graphql [hunk 3]
```diff
@@ -1,57 +1,142 @@
 type OrderOption {
   "The list of concepts and any data on the relationship between this order option and other permitted concepts"
   associationDetails: JSON
+
   "The concept id created by CMR for this orderOption"
   conceptId: String!
+
   "The description for this order option"
   description: String
+
```
Comment (dmistry1) [inline src/types/orderOption.graphql:10]: Can we add `deprecated: String` here.

File: src/datasources/orderOption.js [hunk 4]
```diff
@@ -21,3 +21,43 @@ export default async (params, context, parsedInfo) => {
   // Return a formatted JSON response
   return orderOption.getFormattedResponse()
 }
+
+export const ingestOrderOption = async (args, context, parsedInfo) => {
+  console.log('🚀 ~ ingestOrderOption ~ args:', args)
```
Comment (trevorlang) [inline src/datasources/orderOption.js]: Remove

**Review by abbottry — APPROVED** [review]

**Review by dmistry1 — APPROVED** [review]

File: src/resolvers/__tests__/orderOption.test.js [hunk 5]
```diff
@@ -72,6 +70,7 @@ describe('OrderOption', () => {
       }, {
         contextValue
       })
+      console.log('🚀 ~ test.only ~ response:', response)
```
[the remark this answers is not in the document]
Comment (trevorlang) [reply src/resolvers/__tests__/orderOption.test.js]: Always...

**Pull request merged**